python/foglamp/services/core/api/certificate_store.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import os
+from aiohttp import web
+from foglamp.common import logger
+from foglamp.services.core import connect
+from foglamp.common.configuration_manager import ConfigurationManager
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+_FOGLAMP_DATA = os.getenv("FOGLAMP_DATA", default=None)
+_FOGLAMP_ROOT = os.getenv("FOGLAMP_ROOT", default='/usr/local/foglamp')
+
+
+_logger = logger.setup(__name__, level=20)
+
+_help = """
+    -------------------------------------------------------------------------------
+    | POST             | /foglamp/certificate                                     |
+    | DELETE           | /foglamp/certificate/{name}                              |
+    -------------------------------------------------------------------------------
+"""
+
+
+async def upload(request):
+    """ Upload a certificate
+
+    :Example:
+        curl -F "key=@filename.key" -F "cert=@filename.cert" http://localhost:8081/foglamp/certificate
+        curl -F "key=@filename.key" -F "cert=@filename.cert" -F "overwrite=1" http://localhost:8081/foglamp/certificate
+    """
+    data = await request.post()
+
+    # contains the name of the file in string format
+    key_file = data.get('key')
+    cert_file = data.get('cert')
+
+    # accepted values for overwrite are '0 and 1'
+    allow_overwrite = data.get('overwrite', '0')
+    if allow_overwrite in ('0', '1'):
+        should_overwrite = True if int(allow_overwrite) == 1 else False
+    else:
+        raise web.HTTPBadRequest(reason="Accepted value for overwrite is 0 or 1")
+
+    if not key_file or not cert_file:
+        raise web.HTTPBadRequest(reason="key or certs file is missing")
+
+    key_filename = key_file.filename
+    cert_filename = cert_file.filename
+
+    # accepted extensions are '.key and .cert'
+    valid_extensions = ('.key', '.cert')
+    if not cert_filename.endswith(valid_extensions) or not key_filename.endswith(valid_extensions):
+        raise web.HTTPBadRequest(reason="Accepted file extensions are .key and .cert")
+
+    # certs and key filename should match
+    if cert_filename and key_filename:
+        if cert_filename.split(".")[0] != key_filename.split(".")[0]:
+            raise web.HTTPBadRequest(reason="key and certs file name should match")
+
+    # Get certs directory path
+    certs_dir = _get_certs_dir()
+    found_files = _find_file(cert_filename, certs_dir)
+    is_found = True if len(found_files) else False
+    if is_found and should_overwrite is False:
+        raise web.HTTPBadRequest(reason="Certificate with the same name already exists. "
+                                        "To overwrite set the overwrite to 1")
+
```

Comment:
what if I try to overwrite default or configured certificate, on which this REST api is running currently?!